src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { getUserById } from './utils/getUserById';
+import { getMaxId } from './utils/getMaxId';
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const handleSetTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.trimStart());
+
+    if (hasTitleError) {
+      setHasTitleError(false);
+    }
+  };
+
+  const handleSetUser = (event: ChangeEvent<HTMLSelectElement>) => {
```

Comment:
set or select? 🤔 
i think it will more better
```suggestion
  const handleUserSelection = (event: ChangeEvent<HTMLSelectElement>) => {
```